### PR TITLE
Persist code

### PR DESCRIPTION
**Pull request by tsrines**
Screen Shot 2020-12-03 at 10 59 33 AM

Screen Shot 2020-12-03 at 11 02 18 AM

This PR is a combination of my socket-bugs PR and persist-code.  I had merged them locally before the PR was approved and merged to dev.  

When both users close the interview, instead of clicking end interview, the code will persist and be available on text editor load.

### CODE REVIEW

File: client/src/pages/Interview.js [hunk 1]
```diff
@@ -52,6 +63,9 @@ const Interview = () => {
   const fetchInterview = useCallback(async () => {
     try {
       const { data } = await getInterview(roomId);
+      if (data.interview.code !== '') {
+        setValue(data.interview.code);
```
Comment (ThinkTink) [inline client/src/pages/Interview.js:67]: should this be setOldCode? If not, where is oldCode being set?

File: server/socket/socket.js [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
+      `Connected and authorized, SOCKET ID: ${id} USERID: ${socket.userId}`
+    );
+    socketToUsers[id] = socket.userId;
+    usersToSockets[socket.userId] = socket.id;
+
+    socket.on('leave_interview', ({ roomId }) => {
+      console.log('second person leaving!!!!', Object.keys(rooms[roomId]));
+
+      if (Object.keys(rooms[roomId]).length === 2) {
+        console.log(rooms[roomId].code);
```
Comment (ThinkTink) [inline server/socket/socket.js:28]:
```suggestion
```

File: server/socket/socket.js [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
+    console.log(
+      `Connected and authorized, SOCKET ID: ${id} USERID: ${socket.userId}`
+    );
+    socketToUsers[id] = socket.userId;
+    usersToSockets[socket.userId] = socket.id;
+
+    socket.on('leave_interview', ({ roomId }) => {
+      console.log('second person leaving!!!!', Object.keys(rooms[roomId]));
+
+      if (Object.keys(rooms[roomId]).length === 2) {
```
Comment (ThinkTink) [inline server/socket/socket.js:27]: I'm a bit confused on the logic here. This always logs `second person leaving` but then conditionally logs `first person leaving`? What is `Object.keys(rooms[roomId]).length === 2` checking?

Comment (tsrines) [reply server/socket/socket.js]: the 3rd key is the code.  So if there are only 1 keys, there is only 1 person in the room.  If there are 3 keys, its full, if there is 1 key, there's nobody

File: client/src/pages/Interview.js (hunk 1 above)
Comment (tsrines) [reply client/src/pages/Interview.js:66]: I thought I got rid of oldcode useState.  It's not being set anywhere

**Review by ThinkTink — APPROVED** [review]

**Pull request merged**